querydsl-sql/src/main/java/com/querydsl/sql/MySQLTemplates.java

```diff
@@ -121,7 +121,7 @@ public MySQLTemplates(char escape, boolean quote) {
 
         add(Ops.DateTimeOps.TRUNC_YEAR,   "str_to_date(concat(date_format({0},'%Y'),'-1-1'),'%Y-%m-%d')");
         add(Ops.DateTimeOps.TRUNC_MONTH,  "str_to_date(concat(date_format({0},'%Y-%m'),'-1'),'%Y-%m-%d')");
-        add(Ops.DateTimeOps.TRUNC_WEEK,   "str_to_date(concat(date_format({0},'%Y-%u'),'-2'),'%Y-%u-%w')");
```

Comment:
This always used to truncate to Tuesday. I've changed it to Monday.
[Official documentation](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html) confirms it (see the section on `%u`).

Reply:
Is monday consistant with the start of ISO weeks? Or should that then be sunday?

I vagely remember that most databases follow ISO, where QueryDSL tends to follow the old Java Calendar API, which has a slightly different definition of weeks.

Reply:
According to the docs, it's always going to be Monday (regardless of DB config), which is following ISO standard. I've tested this on several dates and different start of the week setups. 
From what I see, DB are not consistent on following the ISO and some of them even have multiple functions which treat the config differently. The question we should ask is how QueryDSL should attempt to behave?
In my opinion (and my particular use case) it's best to follow ISO regardless of locale. It makes the result predictable and the same across DB. Having solid truncation to Monday is better than inconsisted behaviour based on DB quirks.
I'm open for suggestions, but this is what I'm following currently in this PR (ISO behaviour).